src/kaleidoscope/key_defs.h

```diff
@@ -202,6 +203,45 @@ typedef kaleidoscope::Key Key_;
 #define ___ Key_Transparent
 #define XXX Key_NoKey
 
+namespace kaleidoscope {
+constexpr bool isKeyboardKey(Key key) {
+  return ((key.getFlags() & (RESERVED | SYNTHETIC)) == 0);
+}
+constexpr bool isKeyboardModifier(Key key) {
+  return (isKeyboardKey(key) &&
+          (key.getKeyCode() >= HID_KEYBOARD_FIRST_MODIFIER &&
+           key.getKeyCode() <= HID_KEYBOARD_LAST_MODIFIER));
+}
+constexpr bool isKeyboardShift(Key key) {
```

Comment:
Like "isModifierKey", i think this should be "isShiftKey"

other kinds of shifts that are not referring to keyboard keys should get other names, perhaps ones that don't end in "Key".